src/api/posts.ts

```diff
@@ -1,8 +1,14 @@
 import { client } from '../utils/fetchClient';
 import { Post } from '../types/Post';
 
-export const getUserPosts = (userId: number) => {
-  return client.get<Post[]>(`/posts?userId=${userId}`);
+export const getUserPosts = async (userId: number) => {
+  const res = await client.get<Post[]>(`/posts?userId=${userId}`);
+
+  if (Array.isArray(res)) {
+    return res;
+  }
+
+  throw new Error(`Failed to fetch posts for user ${userId}`);
```

Comment:
The error handling in `getUserPosts` assumes that the response is always an array. Consider checking if the response is an error object or handling different response statuses to ensure robust error handling.